### DISCUSSION TITLE
Filter out BLE from Apple Devices

### DISCUSSION THREAD

**Discussion by Opiekelly** [theengs/gateway#203] [General]
Hello, I would like to filter out all BLE from Apple Devices so my HA is not populated with thousands of devices.  Seems Apple randomizes the MAC addresses and now I have over 7K entries in HA.

**Comment by DigiH** [comment]
Hi, what kind of Apple Devices are you getting?

It could well be that it is the newly introduced `Apple iPhone/iPad` and `Apple Watch` in Theengs Decoder, which are currently not converted to RMACs for filtering in a Theengs Gateway release yet, but will be with [my recent PR](https://github.com/theengs/gateway/pull/187).

You can either [build and install your own build of Theengs Gateway with the latest development branch release](https://gateway.theengs.io/install/install.html#advanced-users-build-and-install), but also rest assured that this will be in the next upcoming Theengs Gateway release.

Generally I would also suggest to turn off discovery so you won't be getting any further HA entries, not just for the Apple devices, but also for any other Theengs compatible BLE devices which might be passing your house.

You can use MQTT Explorer to delete all 7K HA discovery entries with one single procedure ;)

**Reply by DigiH** [reply]
Also, depending on how you have installed Theengs Gateway, you might be able to pip install the previous release of [Theengs Decoder 1.6.4](https://github.com/theengs/decoder/releases/tag/v1.6.4), to alleviate this issue for you until the next Theengs Gateway release.

**Reply by Opiekelly** [reply]
I have turned off discovery.  I will wait for the next build as I have updated the decoder for Aranet4 integration.

**Reply by Opiekelly** [reply]
What would the "topic" look like in MQTT Explorer to delete all the APPLE devices in HA?  I tried the following. 
Publish:
Topic: home/TheengsGateway
payload: empty
retain: true

**Reply by DigiH** [reply]
I'm afraid the issue stems from you having freshly installed Theengs Gateway after the 24th December last year, so it pulled in the then latest release of Theengs Decoder 1.6.7 with the newly introduced Apple Watch, iPhone and iPad decoders, but without the yet released Gateway functionality to filter them, unless you have entered the Identity MAC Address and IRK for your own such devices.

This will automatically resolve with the next release of Gateway.

There is currently no topic or other way to disable this, apart from disabling discovery.

Any particular reason that you cannot keep Discovery disabled until then, once you have all your other owned devices discovered?

And how have you got Gateway installed?

**Reply by Opiekelly** [reply]
Hello, so I cleared out the binary_sensor entries and disabled discovery.  However when I start the gateway the binary_sensors start to populate entries with APPLEDEVICE-unlocked and HA MQTT devices start growing.  This is with the current addon release 1.2.0.

`
MQTT_HOST: 192.168.0.128
MQTT_USERNAME: homeassistant
MQTT_PASSWORD: password
MQTT_PORT: 1883
MQTT_PUB_TOPIC: home/TheengsGateway/BTtoMQTT
MQTT_SUB_TOPIC: home/+/BTtoMQTT/undecoded
MQTT_PRE_TOPIC: home/presence/TheengsGateway
PRESENCE: false
PUBLISH_ALL: true
PUBLISH_ADVDATA: true
SCAN_DUR: 10
TIME_BETWEEN: 60
LOG_LEVEL: WARNING
DISCOVERY: false
DISCOVERY_TOPIC: homeassistant
DISCOVERY_DEVICE_NAME: TheengsGateway
DISCOVERY_FILTER: "[IBEACON]"
ADAPTER: ""
SCANNING_MODE: passive
TIME_SYNC: "[]"
TIME_FORMAT: false
ENABLE_TLS: false
ENABLE_WEBSOCKET: false
IDENTITIES: ""
BINDKEYS: ""
`

**Reply by DigiH** [reply]
Please add APPLEDEVICE and APPLEWATCH to the DISCOVERY_FILTER, so that it will be

`[IBEACON APPLEDEVICE APPLEWATCH] `

This should stop all new discoveries.

**Comment by Opiekelly** [comment]
I am just trying to clean up the old discovered devices via MQTT explorer as you mentioned.  Just figured I would ask how to get that done. On Jan 6, 2024, at 16:35, DigiH ***@***.***> wrote:﻿
I'm afraid the issue stems from you having freshly installed Theengs Gateway after the 24th December last year, so it pulled in the then latest release of Theengs Decoder 1.6.7 with the newly introduced Apple Watch, iPhone and iPad decoders, but without the yet released Gateway functionality to filter them, unless you have entered the Identity MAC Address and IRK for your own such devices.
This will automatically resolve with the next release of Gateway.
There is currently no topic or other way to disable this, apart from disabling discovery.
Any particular reason that you cannot keep Discovery disabled until then, once you have all your other owned devices discovered?

**Reply by DigiH** [reply]
In MQTT Explorer expand the `homeassistance` topic, and there the `binary_sensor` sub topic.

You can delete any `-unlocked` entries there. 

Depending on how many entries there are you might want to delete the whole binary_sensor sub topic, turn on Discovery again until your potential own other `binary_sensor` devices are discovered again, then turn it off again.

Have a look at what other than `-unlocked` entries are there in case of deleting the whole sub topic.
